twitterpuller/src/main/java/de/brandwatch/minianalytics/twitterpuller/twitter/TweetSearcher.java

```diff
@@ -0,0 +1,53 @@
+package de.brandwatch.minianalytics.twitterpuller.twitter;
+
+import de.brandwatch.minianalytics.twitterpuller.kafka.Producer;
+import de.brandwatch.minianalytics.twitterpuller.model.Resource;
+import de.brandwatch.minianalytics.twitterpuller.postgres.cache.QueryCache;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import twitter4j.*;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+
+@Component
+public class TweetSearcher {
+
+    @Autowired
+    QueryCache queryCache;
+
+    @Autowired
+    Producer producer;
+
+    private static final Twitter twitter = TwitterFactory.getSingleton();
+
+    @Scheduled(fixedDelay = 1000)
+    public void searchTweets(){
```

Comment:
Sometimes it's good to extract methods to improve readability. For example here, I would suggest something like this:

```java
@Scheduled(fixedDelay = 1000)
public void searchTweets(){
    queryCache.getCachedQueries()
            .forEach(this::search);
}

private void search(de.brandwatch.minianalytics.twitterpuller.postgres.model.Query query) {
    try {
        Query searchQuery = new Query(query.getQuery());
        searchQuery.setCount(100);

        QueryResult queryResult = twitter.search(searchQuery);

        queryResult.getTweets().stream()
                .map(this::mapToResource)
                .forEach(resource -> producer.send(resource));

    } catch (TwitterException e) {
        e.printStackTrace();
    }
}

private Resource mapToResource(Status tweet) {
    Instant date = new Timestamp(tweet.getCreatedAt().getTime()).toInstant();

    Resource resource = new Resource();
    resource.setDate(date);
    resource.setText(tweet.getText());
    resource.setAuthor(tweet.getUser().getName());
    return resource;
}
```

What do you think?

Reply:
I actually thought that the readability was quite good because of the lambdas. But I don't mind breaking it into different methods

Reply:
It gets easier because for example if I want to understand the `search` method, I don't necessarily need to know the details of mapping tweet to resource - I can tell just by the method name what's happening. That way, I can grasp the overall logic much quicker. Very long methods are not desirable. I didn't check, but I reckon somewhere in Clean Code they talk about that 😄 Did you check that book out?

Reply:
I googled it and read the part about Naming class, variables etc.